packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -0,0 +1,444 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:developer' as developer;
+import 'dart:math' as math;
+import 'dart:ui' as ui show window;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/scheduler.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/material.dart';
+import 'package:vector_math/vector_math_64.dart';
+
+import 'basic.dart';
+import 'binding.dart';
+import 'framework.dart';
+import 'gesture_detector.dart';
+
+/// A widget that enables inspecting the child widget's structure.
+///
+/// This widget is useful for understand how an app is structured enabling
+/// interactive
+class WidgetInspector extends StatefulWidget {
+  /// Creates a widget that enables inspection for the child.
+  ///
+  /// The [child] argument must not be null.
+  const WidgetInspector({Key key, this.child}) : super(key: key);
+
+  /// The widget below this widget in the tree.
+  final Widget child;
+
+  @override
+  _WidgetInspectorState createState() => new _WidgetInspectorState();
+}
+
+class _WidgetInspectorState extends State<WidgetInspector>
+    with WidgetsBindingObserver {
+  _WidgetClient _client;
+
+  Offset _lastPointerLocation;
+
+  /// Selected [RenderObject] being inspected.
+  ///
+  /// Only RenderObjects that are RenderBox instances are currently supported.
+  RenderBox _selected;
+
+  /// Whether the inspector is in select mode.
+  ///
+  /// In select mode pointer interactions trigger widget selection instead of
+  /// normal interactions.
+  /// Not in select mode, the selected widget is highlighted but the application
```

Comment:
Paragraph breaks require a blank line. If you meant this to be three paragraphs, you need a new newline here. If you meant it to be two, you should probably rewrap the second paragraph to have consistent line lengths.